xc/xenops_server_xen.ml

```diff
@@ -263,9 +263,11 @@ let params_of_backend backend =
     | [] ->
       raise (Xenopsd_error (Internal_error ("Could not find XenDisk implementation: " ^ (Storage_interface.(rpc_of backend) backend |> Jsonrpc.to_string))))
   in
-  let params, extra_keys = match blockdevs, files, nbds, xendisks with
-  | {path}::_, _, _, _ | _, {path}::_, _, _ -> (path, [])
-  | _, _, {uri}::_, xendisk::_ -> (uri, ["qemu-params", xendisk.Storage_interface.params])
+  let params, extra_keys = match prefer_nbd, blockdevs, files, nbds, xendisks with
+  | true,_, _, {uri}::_, xendisk::_ -> (uri, ["qemu-params", xendisk.Storage_interface.params])
+  | true,{path}::_, _, _, _ | true,_, {path}::_, _, _ -> (path, [])
+  | false,{path}::_, _, _, _ | false,_, {path}::_, _, _ -> (path, [])
```

Comment:
When path is first you can merge the true and false branches by replacing these 2 lines with:
```ocaml
_,{path}::_, _, _, _ | _, _, {path}::_, _, _ ->
```